modules/javafx.graphics/src/main/java/com/sun/javafx/application/preferences/PlatformPreferences.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.application.preferences;
+
+import com.sun.javafx.binding.MapExpressionHelper;
+import javafx.application.Appearance;
+import javafx.application.Platform;
+import javafx.beans.InvalidationListener;
+import javafx.beans.property.ReadOnlyObjectProperty;
+import javafx.collections.MapChangeListener;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Contains the implementation of a read-only map of platform preferences.
+ * <p>
+ * When the operating system signals that a preference has changed, the mappings are updated
+ * by calling the {@link #update(Map)} method.
+ */
+public class PlatformPreferences extends AbstractMap<String, Object> implements Platform.Preferences {
+
+    /**
+     * Contains mappings from platform-specific keys to well-known keys, which are used
+     * in the implementation of the property-based API in {@link PreferenceProperties}.
+     */
+    final Map<String, String> wellKnownKeys;
+    final Map<String, Object> effectivePreferences = new HashMap<>();
+    final Map<String, Object> unmodifiableEffectivePreferences = Collections.unmodifiableMap(effectivePreferences);
+    final PreferenceProperties properties = new PreferenceProperties(this);
```

Comment:
minor: `this` escapes here before object is fully constructed

Reply:
Yes, but the `PreferenceProperties` doesn't to anything with `this`.